### PR TITLE
implemented %c, %g and %p classes

### PR DESCRIPTION
**Pull request by NotJustAToy**
Thank you for you hard work and this great package!
I try implement the following classes: %c, %g and %p.

### CODE REVIEW

**Comment by jcgoble3** [comment]
Wow, I had forgotten about this project. As a preliminary step, I have updated the .travis.yml file to test modern versions of Python. I have not looked at any of your code yet and will do so within the next few days (I'm currently very busy with schoolwork).

**Comment by jcgoble3** [comment]
Looks like the Travis config needs a bit more updating than I thought. Give me some time to work it out; I haven't used Travis in a while.

**Comment by jcgoble3** [comment]
Tests are now working again. I will schedule time to look at the code this weekend.

**Comment by NotJustAToy** [comment]
Thank you!

**Comment by jcgoble3** [comment]
Out of curiosity, what do you use this package for? I wasn't aware that anyone actually used it. I created this mainly as a way to familiarize myself with Lua's pattern-matching code (written in C) and for my own amusement. I published it on PyPI only because it was reasonably complete, but didn't expect anyone to actually use it.

File: src/luapatt.py [hunk 1]
```diff
@@ -19,8 +19,9 @@
 # IN THE SOFTWARE.
 
 from array import array
+import unicodedata
 
-__version__ = '0.9.0b5'
+__version__ = '0.9.0b6'
```
Comment (jcgoble3) [inline src/luapatt.py]: The version change should be reverted. I'll handle the version update myself as part of the release process, and the next release might be something different (such as 1.0).

```suggestion
__version__ = '0.9.0b5'
```

File: tests/test_lua1_basics.py [hunk 2]
```diff
@@ -162,3 +162,9 @@ def test_right_bracket_in_set():
     assert luapatt.match(']]]áb', '[^]]') == 'á'
 def test_percent_x():
     assert luapatt.match("0alo alo", "%x*") == "0a"
+def test_match_control_characters():
+    assert luapatt.match('alo alo', '%C+') == 'alo alo'
+def test_match_printable():
+    assert luapatt.match('  \n\r*&\n\r   xuxu  \n\n', '%g%g%g+') == 'xuxu'
```
Comment (jcgoble3) [inline tests/test_lua1_basics.py:168]: I was going to insist on using the tests from the official Lua test suite, but you've already done that. Nice!

File: tests/test_lua1_basics.py [hunk 3]
```diff
@@ -162,3 +162,9 @@ def test_right_bracket_in_set():
...
+def test_match_punctuation():
+    assert luapatt.match('Hello World!', '%p+') == '!'
```
Comment (jcgoble3) [inline tests/test_lua1_basics.py:170]: This one doesn't seem to be tested in the Lua test suite, so I'm satisfied with this test for now.

File: src/luapatt.py [hunk 4]
```diff
@@ -443,9 +444,12 @@ def matchclass(self, sc, pc):
             match = sc.isdigit() or sc in 'abcdefABCDEF'
         elif pcl == 'z':
             match = sc == '\0'
-        elif pcl in 'cgp':
-            raise NotImplementedError('{0}c, {0}g, and {0}p are not '
-                                      'available'.format(ESCAPE))
+        elif pcl == 'c':
+            match = unicodedata.category(sc)[0] == 'C'
+        elif pcl == 'g':
+            match = unicodedata.category(sc)[0] not in ('C', 'Z')
```
Comment (jcgoble3) [inline src/luapatt.py]: I don't like tuples of single characters. Please change the tuple to a string (like in the removed line 446).

```suggestion
            match = unicodedata.category(sc)[0] not in 'CZ'
```

**Review by jcgoble3 — CHANGES_REQUESTED** [review]
Just two minor things in the code to alter before I'm ready to merge. Check the suggested changes in the review; they're minor enough that if you don't make the edits by tomorrow afternoon/evening (US time), I'll make them myself, but I want to give you the chance to do it first since it's your contribution.

I would also like to see a README update since this renders the first bullet point obsolete and applies them in a different way from what the second bullet point mentions. Again, if I don't get anything from you by tomorrow evening, I'll probably do this myself.

Also, as mentioned above, I'm interested to hear how you're using this. If people are legitimately using this, I'll clean it up a tad and get a real 1.0 release out, and possibly add some additional features depending on my available time.

**Comment by jcgoble3** [comment]
Apologies for the commit noise. I pushed the .travis.yml updates to my own develop branch and tweaked the README to change the minimum supported version of Python, and then GitHub complained about branches being out of sync, so I had to merge those into your branch. When this is ready to merge, I'll do a squash merge, so all these excess merge commits and stuff will go away and we'll end up with one nice, neat commit.

**Comment by NotJustAToy** [comment]
I wrote a lua interpreter in python and want to use this package in the implementation of the standard string library. I already tested it on `pm.lua` from the stock test suite and all tests pass.

**Review by jcgoble3 — APPROVED** [review]

**Comment by jcgoble3** [comment]
A Lua interpreter in Python? Interesting. I've thought about doing the same, but never got around to actually starting it. Do you have the code open source? I'm interested to look at it if you do.

Anyway, everything looks good, and the check for `Mapping` instead of `dict` is good too. I'll merge this in a moment.

**Comment by NotJustAToy** [comment]
I think I can show the code in a couple of weeks. I will definitely inform you about this. Thanks!

**Pull request merged**